pkg/nl/layer2.go

```diff
@@ -327,6 +363,48 @@ func (n *NetlinkManager) ListL2() ([]Layer2Information, error) {
 	return n.listL2()
 }
 
+func (n *NetlinkManager) ListNeighbors() ([]NeighborInformation, error) {
+	netlinkNeighbors, err := n.listNeighbors()
+	if err != nil {
+		return nil, err
+	}
+	neighbors := []NeighborInformation{}
+	for _, netlinkNeighbor := range netlinkNeighbors {
+		family, err := GetFamily(netlinkNeighbor.Family)
+		if err != nil {
+			return nil, err
+		}
+		state, err := getNeighborState(netlinkNeighbor.State)
+		if err != nil {
+			return nil, err
+		}
+
+		linkInfo, err := netlink.LinkByIndex(netlinkNeighbor.LinkIndex)
+		if err != nil {
+			return nil, err
```

Comment:
should I complete the loop and then return how should this be done properly?

Reply:
Starting with 1.13 go has ability to create error hierarchy by 'wrapping' them. This way you're able to give more context to the error. You might see complete explanation here: https://go.dev/blog/go1.13-errors

For what you should do is, to instead of returning `err` directly, just return something like

```
return nil, fmt.Errorf("description of the error: %w", err)
```

For example here it could be:

```
return nil, fmt.Errorf("error getting link by index: %w", err)
```

where _%w_ is format parameter that indicates error to be wrapped.